management-interface.go

```diff
@@ -39,21 +44,82 @@ func showDiskSpace(w http.ResponseWriter) error {
 		fmt.Fprintf(w, "Cannot show disk space at this time.\n")
 		return err
 	}
-	fmt.Fprintf(w, "%s\n", out)
+	fmt.Fprintf(w, "Disk space usage is: \n\n%s\n", out)
 	return nil
 
 }
 
+// Show the available network interfaces on this machine.
+func availableInterfaces(w http.ResponseWriter) error {
+
+	interfaces, err := net.Interfaces()
+
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(w, "Available network interfaces on this machine:\n\n")
+	for _, i := range interfaces {
+		if i.Name != "lo" {
+
+			fmt.Fprintf(w, "Testing %v...\n", i.Name)
+			fmt.Fprintf(w, "  Sending %v packets\n", NumPacketsToSend)
+
+			// Windows
+			//commandStr := "ping -n " + strconv.Itoa(NumPacketsToSend) + " -w 15 1.1.1.1"
+			//out, _ := exec.Command("cmd", "/C", commandStr).Output()
+			//pos := strings.Index(outStr, "Received =")
+			//if pos != -1 {
+			//	numPacketsReceivedStr := string(outStr[pos+11])
+
+			// Unix
+			commandStr := "ping -I " + i.Name + " -c " + strconv.Itoa(NumPacketsToSend) + " -n -W 15 1.1.1.1"
+			out, _ := exec.Command("sh", "-c", commandStr).Output()
+
+			// Did we get the output we'd expect if the network interface is up?
+			interfaceUp := false
+			outStr := string(out)
+			pos := strings.Index(outStr, "transmitted")
+			if pos != -1 {
+				numPacketsReceivedStr := string(outStr[pos+13])
+				fmt.Fprintf(w, "  Received %v packets\n", numPacketsReceivedStr)
+				numPacketsReceived, _ := strconv.Atoi(numPacketsReceivedStr)
+				if numPacketsReceived > 0 {
+					// I consider this interface to be "up".
+					interfaceUp = true
+				}
+			}
```

Comment:
Instead of parsing the output from the ping command you can check the exit code. Here's the relevant bit from the ping man page:

> If ping does not receive any reply packets at all it will exit with code 1. If a packet count and deadline are  both specified,  and  fewer  than count packets are received by the time the deadline has arrived, it will also exit with code 1.  On other error it exits with code 2. Otherwise it exits with code 0. This makes it possible to use the exit code to see if a host is alive or not.

There will be an error from `Output()` (which is currently being ignored) when the ping tests fail.